File: Baekjoon/Class2/25.1920 수 찾기.py
import bisect
import sys
input = sys.stdin.readline
write = sys.stdout.write

def bin_search(li, target):
    left_insert = bisect.bisect_left(li, target)
    if left_insert < len(li) and li[left_insert] == target:
        return True
    return False

def main():
    _ = input()
    n_list = sorted(map(int, input().split())) #정렬
    _ = input()
    m_list = list(map(int, input().split()))

    for m_num in m_list:
        #이진 탐색 O(logn)
        if bin_search(n_list, m_num):
            write("1\n")
        else:
            write("0\n")

main()

''' 이진 탐색 구현
import sys
input = sys.stdin.readline
write = sys.stdout.write

def bin_search(li, target):
    left = 0
    right = len(li) - 1
    while left <= right:
        middle = (left + right) // 2
        if li[middle] == target:
            return middle
        elif li[middle] < target:
            left = middle + 1
        else:
            right = middle - 1
    return -1

def main():
    _ = input()
    n_list = sorted(map(int, input().split())) #정렬
    _ = input()
    m_list = list(map(int, input().split()))

    for m_num in m_list:
        #이진 탐색 O(logn)
        if bin_search(n_list, m_num) > -1:
            write("1\n")
        else:
            write("0\n")

main()
'''

''' set 사용
import sys
input = sys.stdin.readline
write = sys.stdout.write

_ = input()
n_set = set(map(int, input().split()))
_ = input()
m_list = list(map(int, input().split()))

for m_num in m_list:
    #set는 해시 테이블을 사용하므로 in 연산에 대해 O(1)
    if m_num in n_set:
        write("1\n")
    else:
        write("0\n")
'''

File: Baekjoon/Class2/25.1920 수 찾기.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const nSet = new Set(input[1].split(" ").map(Number));
const mArr = input[3].split(" ").map(Number);
const out = [];

for(const mNum of mArr) {
    // Set는 해시 테이블을 사용하므로 has()에 대해 O(1)
    if(nSet.has(mNum))
        out.push("1");
    else
        out.push("0");
}
console.log(out.join("\n"));

/* 이진 탐색 구현
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const binSearch = (arr, target) => {
    let left = 0, right = arr.length - 1;
    while (left <= right) {
        let middle = Math.trunc((left + right) / 2);
        switch(Math.sign(arr[middle] - target)) {
            case 0: return middle;
            case -1: left = middle + 1;
                break;
            case 1: right = middle - 1;
        }
    }
    return -1;
}

const nArr = input[1].split(" ").map(Number).sort((a, b) => a - b); // 정렬
const mArr = input[3].split(" ").map(Number);
const out = [];

for(const mNum of mArr) {
    // 이진 탐색 O(logn)
    if(binSearch(nArr, mNum) > -1)
        out.push("1");
    else
        out.push("0");
}
console.log(out.join("\n"));
*/